controller.py

```diff
@@ -290,6 +291,10 @@ def controller(
             (config_qemu["max_instruction_count"], goldenrun_data, faultlist),
             lzma.open("bkup_goldenrun_results.xz", "wb"),
         )
+
+        clogger.info(f"Got {len(faultlist)} fault entries")
+        if goldenrun_only:
+            return config_qemu
```

Comment:
If we start a new simulation with the goldenrun backup file we created using the `--goldenrun-only` flag, the goldenrun data is not written to the output HDF5 file. We should detect this and write the data from the backup file into the HDF5 file.